### ISSUE TITLE
No way to disable shared URLSession

### ISSUE THREAD

**Issue by TaborKelly** [IBM-Swift/SwiftyRequest#65]
There is no way to disable the shared URLSession:
```swift
public class RestRequest: NSObject  {
...
    /// A default `URLSession` instance.
    #if swift(>=4.1)
    private var session: URLSession = URLSession.shared
    #else
    private var session: URLSession = URLSession(configuration: URLSessionConfiguration.default)
    #endif
...
    public init(method: HTTPMethod = .get, url: String, containsSelfSignedCert: Bool? = false, clientCertificate: ClientCertificate? = nil) {
...
```

**Comment by Andrew-Lees11** [comment]
Do you require it to not be a shared session?
 We could allow you to provide your own session so you can set configuration values if you need.

**Comment by TaborKelly** [comment]
I'm using SwiftyRequest for testing an existing Kitura API that is designed for access by vanilla JavaScript. For some of my test cases I need to be able to reset the cookie store. After further investigation, what I really need to be able to do is `reset()` the `URLSession` that I am using as `finishTasksAndInvalidate()` doesn't seem to be enough.

Which is a long way of saying, yet, if I could provide my own `URLSession` that would be very helpful.

**Comment by Andrew-Lees11** [comment]
@TaborKelly 
Can you just confirm that pr #66 making the URLSession public would fix your issue?

**Comment by Andrew-Lees11** [comment]
Closing since the URLSession is now public. This will give you access to the session change it or replace it and has been released in version 2.1.0

**Comment by TaborKelly** [comment]
Sorry for the late reply. Yes, that looks great!
